### DISCUSSION TITLE
[General] How to know ZkSync transaction has been confirmed or not at Layer 1

### DISCUSSION THREAD

**Discussion by farhanhilmi** [zkSync-Community-Hub/zksync-developers#559] [General] [answered]
### Environment

Mainnet

Haii, I want to find out if there is an endpoint that can see the L1 Transaction hash of my ZkSync transaction batch submitted to L1? I need to know this because I want to make sure the transaction has been confirmed on L1.  Something similar to this Arb maybe  https://github.com/OffchainLabs/arbitrum-tutorials/tree/master/packages/l1-confirmation-checker

Thanks 🎉

**Comment by Praizl** [comment]
The only way I know to check execution on Layer 1 is through explorers. I don’t think they have shared an endpoint for tracking this yet

**Comment by 0xsisyfos** [comment, marked as answer]
Hey @farhanhilmi 
For a given transaction hash, with the Web3 JSON-RPC API, you can more details on the transaction including when it has been committed/proven/executed: https://docs.zksync.io/build/api.html#zks-gettransactiondetails

zkSync Batches will go through the steps above (committed, proven, executed), and each of them have a corresponding L1 hash. The transaction is considered final, once it has passed the 'executed" stage (`ethProveTxHash` in the API call above)

**Reply by farhanhilmi** [reply]
Thanks mate. This is very helpful 🎉
